README.md

```diff
@@ -48,9 +48,9 @@ The WAF bindings have multiple moving parts that are necessary to understand:
 - Handle: a object wrapper over the pointer to the C WAF Handle
 - Context: a object wrapper over a pointer to the C WAF Context
 - Encoder: whose goal is to construct a tree of Waf Objects to send to the WAF
-- Allocator: Does all writing and allocation operations for the construction of Waf Objects
+- CGORefPol: Does all allocation operations for the construction of Waf Objects and keep track of the equivalent go pointers
 - Decoder: Transforms Waf Objects returned from the WAF to usual go objects (e.g. maps, arrays, ...)
-- Library: The library which wraps all calls to C code
+- Library: The low-level API of bindings with the C world, made to provide Go type checking to it
```

Comment:
```suggestion
- Library: The low-level go bindings to the C library, providing improved typings to go consumers
```

Reply:
I'm saying these make it easier to code against the C code in go... consumers can be internal, too.

Reply:
```suggestion
- Library: The low-level go bindings to the C library, providing improved typing
```